components/buttons/MainButton/MainButton.tsx

```diff
@@ -0,0 +1,93 @@
+'use client';
+import { ReactNode } from 'react';
+
+interface MainButtonProps {
+  children: ReactNode;
+  clickHandler: () => void;
+  variant: 'mobile' | 'desktop';
+  size: 'small' | 'large';
+  colour: 'primary' | 'secondary' | 'tertiary';
+  lightMode: boolean;
+  type: 'button' | 'submit' | 'reset';
+  disabled: boolean;
+  ariaLabel?: string;
```

Comment:
@nichgalzin Good work! The structure is clear and easy to follow, with well-chosen variable names.

I have a small suggestion: since the buttons like Delete, Cancel, and Remove follow a black-and-white design, we could add a purpose prop to specify the button's role (e.g., 'cancel', 'delete', or 'remove') and extend the `colourStyles` logic to make it more flexible. There are different ways to approach this, so it’s totally up to you.

Also, you’re spot on for asking the designers in the Figma file about adding focus and hover styles.